lib/factory_bot.rb

```diff
@@ -109,36 +113,7 @@ class << self
               :traits,
               deprecator: Deprecation
   end
-
-  def self.register_strategy(strategy_name, strategy_class)
-    strategies.register(strategy_name, strategy_class)
-    StrategySyntaxMethodRegistrar.new(strategy_name).define_strategy_methods
-  end
-
-  def self.strategy_by_name(name)
-    strategies.find(name)
-  end
-
-  def self.register_default_strategies
```

Comment:
We should delegate and deprecate `register_default_strategies` and `register_default_callbacks` too